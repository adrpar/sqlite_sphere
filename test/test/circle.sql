.load ../build/libsqlitesphere.so

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 1.0d >'), "RAD", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 1d 30m >'), "RAD", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 0.1 >'), "RAD", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 90d >'), "RAD", 8);


SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 1.0d >'), "DEG", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 1d 30m >'), "DEG", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 0.1 >'), "DEG", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 90d >'), "DEG", 8);


SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 1.0d >'), "DMS", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 1d 30m >'), "DMS", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 0.1 >'), "DMS", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 90d >'), "DMS", 8);


SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 1.0d >'), "HMS", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 1d 30m >'), "HMS", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 0.1 >'), "HMS", 8);

SELECT sstr(scircle( '< (1h 2m 30s , +1d 2m 30s), 90d >'), "HMS", 8);




-- Functions


-- checking spherical circle operators

SELECT sequal( scircle( '<(0, 90d),1>' )  ,  scircle( '<(0, 90d),1>' ));
SELECT snotequal( scircle( '<(0,-90d),1>') , scircle( '<(0, 90d),1>' ));


-- float8 sdist(scircle,scircle)

SELECT 180.0*sdist(scircle( '<( 0h 2m 30s , 10d 0m 0s), 0.1d>'),scircle( '<( 0h 2m 30s , -10d 0m 0s),0.1d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 0m 00s , 0d 0m 0s), 1.0d>'),scircle( '<( 1h 0m 00s , 0d 0m 0s),1.0d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 23h 30m 00s , 0d 0m 0s), 1.0d>'),scircle( '<( 1h 0m 00s , 0d 0m 0s),1.5d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 40m 00s , 0d 0m 0s), 1.0d>'),scircle( '<( 0h 50m 00s , 0d 0m 0s),1.0d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 40m 00s , 0d 0m 0s), 1.5d>'),scircle( '<( 0h 50m 00s , 0d 0m 0s),1.5d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 40m 00s , 90d 0m 0s), 1.5d>'),scircle( '<( 0h 50m 00s , 0d 0m 0s),1.5d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 40m 00s , 90d 0m 0s), 1.0d>'),scircle( '<( 0h 50m 00s , -90d 0m 0s),1.0d>'))/pi();



-- spoint scenter(scircle)

SELECT sstr(scenter(scircle( '< (1h 2m 30s , +1d 2m 30s), 1.0d >')), "DMS", 8);

SELECT sstr(scenter(scircle( '< (1h 2m 30s , +90d 0m 0s), 1.0d >')), "DMS", 8);



-- scircle scircle(spoint)

SELECT sstr(scircle(spoint( '(0d,0d)' )), "DMS", 8);

SELECT sstr(scircle(spoint( '(0d,90d)' )), "DMS", 8);

SELECT sstr(scircle(spoint( '(0d,-90d)' )), "DMS", 8);



-- Operators

-- ,  opsequal( erato)r

-- should be "true"

SELECT sequal( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,+1d), 1.5d >') );

SELECT sequal( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (25h 0m 0s,+1d), 1d 30m >') );

SELECT sequal( scircle( '< (1h 0m 0s,+95d), 1.5d >'), scircle( '< (13h 0m 0s,+85d), 1.5d >') );

SELECT sequal( scircle( '< (1h 0m 0s,-95d), 1.5d >'), scircle( '< (13h 0m 0s,-85d), 1.5d >') );

SELECT sequal( scircle( '< (1h 0m 0s,+90d), 1.5d >'), scircle( '< (2h 0m 0s,+90d), 1.5d >') );

SELECT sequal( scircle( '< (1h 0m 0s,-90d), 1.5d >'), scircle( '< (2h 0m 0s,-90d), 1.5d >') );

-- should be "false"

SELECT sequal( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,-1d), 1.5d >') );

SELECT sequal( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,+1d), 2.5d >') );



-- osnotequal( perato9r

-- should be "false"

SELECT snotequal( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,+1d), 1.5d >') );

SELECT snotequal( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (25h 0m 0s,+1d), 1d 30m >') );

SELECT snotequal( scircle( '< (1h 0m 0s,+95d), 1.5d >'), scircle( '< (13h 0m 0s,+85d), 1.5d >') );

SELECT snotequal( scircle( '< (1h 0m 0s,-95d), 1.5d >'), scircle( '< (13h 0m 0s,-85d), 1.5d >') );

SELECT snotequal( scircle( '< (1h 0m 0s,+90d), 1.5d >'), scircle( '< (2h 0m 0s,+90d), 1.5d >') );

SELECT snotequal( scircle( '< (1h 0m 0s,-90d), 1.5d >'), scircle( '< (2h 0m 0s,-90d), 1.5d >') );

-- should be "true"

SELECT snotequal( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,-1d), 1.5d >') );

SELECT snotequal( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,+1d), 2.5d >') );



-- osoverlaps(perato )r

-- should be "true"

SELECT soverlaps( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,+1d), 1.5d >') );

SELECT soverlaps( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,+1d), 0.5d >') );

SELECT soverlaps( scircle( '< (1h 0m 0s,+1d), 1.5d >'), scircle( '< (1h 0m 0s,-1d), 1.5d >') );

SELECT soverlaps( scircle( '< (1h 0m 0s,0d), 15d >'), scircle( '< (0h 0m 0s,0d), 15d >') );

SELECT soverlaps( scircle( '< (2h 0m 0s,0d), 15d >'), scircle( '< (0h 0m 0.1s,0d), 15d >') );

SELECT soverlaps( scircle( '< (1h 0m 0s,0d), 15d >'), scircle( '< (23h 0m 0.1s,0d), 15d >') );

-- should be "false"

SELECT soverlaps( scircle( '< (1h 0m 0s,+1d), 0.5d >'), scircle( '< (1h 0m 0s,-1d), 0.5d >') );

SELECT soverlaps( scircle( '< (1d 0m 0s,+1d), 1.0d >'), scircle( '< (0d 0m 0s,0d), 0.1d >') );



-- scenter( operato )r

SELECT sstr(scenter( scircle( '< (1h 2m 3s , +1d 2m 3s), 1.0d >') ), "DMS", 8);

SELECT sstr(scenter( scircle( '< (1h 2m 3s , +90d 0m 0s), 1.0d >') ), "DMS", 8);



-- ,  sdist( operato)r

SELECT 180.0*sdist(scircle( '<( 0h 2m 30s , 10d 0m 0s), 0.1d>'), scircle( '<( 0h 2m 30s , -10d 0m 0s),0.1d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 0m 00s , 0d 0m 0s), 1.0d>'), scircle( '<( 1h 0m 00s , 0d 0m 0s),1.0d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 23h 30m 00s , 0d 0m 0s), 1.0d>'), scircle( '<( 1h 0m 00s , 0d 0m 0s),1.5d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 40m 00s , 0d 0m 0s), 1.0d>'), scircle( '<( 0h 50m 00s , 0d 0m 0s),1.0d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 40m 00s , 0d 0m 0s), 1.5d>'), scircle( '<( 0h 50m 00s , 0d 0m 0s),1.5d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 40m 00s , 90d 0m 0s), 1.5d>'), scircle( '<( 0h 50m 00s , 0d 0m 0s),1.5d>'))/pi();

SELECT 180.0*sdist(scircle( '<( 0h 40m 00s , 90d 0m 0s), 1.0d>'), scircle( '<( 0h 50m 00s , -90d 0m 0s),1.0d>'))/pi();



-- scirsrcontainsl( cle, scircle operato )r

-- should be "true"

SELECT srcontainsl(  scircle( '< (1h 0m 0s,+1d),  0.5d >'), scircle( '< (1h 0m 0s,+1d),  1.5d >') );

SELECT srcontainsl(  scircle( '< (2d 0m 0s,+1d),  0.5d >'), scircle( '< (1d 0m 0s,0d),   3.5d >') );

SELECT srcontainsl(  scircle( '< (1h 0m 0s,+89d), 0.5d >'), scircle( '< (1h 0m 0s,+90d), 1.5d >') );

SELECT srcontainsl(  scircle( '< (1h 0m 0s,-89d), 0.5d >'), scircle( '< (1h 0m 0s,-90d), 1.5d >') );

-- should be "false"

SELECT srcontainsl(  scircle( '< (1h 0m 0s,+1d),  0.5d >'), scircle( '< (2h 0m 0s,+1d),  1.5d >') );

SELECT srcontainsl(  scircle( '< (2d 0m 0s,+1d),  1.5d >'), scircle( '< (0d 0m 0s, 0d),  3.5d >') );

SELECT srcontainsl(  scircle( '< (1h 0m 0s,+89d), 0.5d >'), scircle( '< (1h 0m 0s,+90d), 1.0d >') );

SELECT srcontainsl(  scircle( '< (1h 0m 0s,-89d), 0.5d >'), scircle( '< (1h 0m 0s,-90d), 1.0d >') );



-- scirslcontainsr( cle, scircle operato )r

-- should be "true"

SELECT slcontainsr( scircle( '< (1h 0m 0s,+1d),  1.5d >') , scircle( '< (1h 0m 0s,+1d),  0.5d >') );

SELECT slcontainsr( scircle( '< (1d 0m 0s,0d),   3.5d >') , scircle( '< (2d 0m 0s,+1d),  0.5d >') );

SELECT slcontainsr( scircle( '< (1h 0m 0s,+90d), 1.5d >') , scircle( '< (1h 0m 0s,+89d), 0.5d >') );

SELECT slcontainsr( scircle( '< (1h 0m 0s,-90d), 1.5d >') , scircle( '< (1h 0m 0s,-89d), 0.5d >') );

-- should be "false"

SELECT slcontainsr( scircle( '< (2h 0m 0s,+1d),  1.5d >') , scircle( '< (1h 0m 0s,+1d),  0.5d >') );

SELECT slcontainsr( scircle( '< (0d 0m 0s, 0d),  3.5d >') , scircle( '< (2d 0m 0s,+1d),  1.5d >') );

SELECT slcontainsr( scircle( '< (1h 0m 0s,+90d), 1.0d >') , scircle( '< (1h 0m 0s,+89d), 0.5d >') );

SELECT slcontainsr( scircle( '< (1h 0m 0s,-90d), 1.0d >') , scircle( '< (1h 0m 0s,-89d), 0.5d >') );



-- spoisrcontainsl( nt, scircle operato )r

-- should be "true"

SELECT srcontainsl(  spoint( '(1h 0m 0s,+1d) ' ), scircle( '< (1h 0m 0s,+1d),  1.5d >') );

SELECT srcontainsl(  spoint( '(2d 0m 0s,+1d) ' ), scircle( '< (1d 0m 0s, 0d),  3.5d >') );

SELECT srcontainsl(  spoint( '(1h 0m 0s,+89d)' ), scircle( '< (1h 0m 0s,+90d), 1.5d >') );

SELECT srcontainsl(  spoint( '(1h 0m 0s,-89d)' ), scircle( '< (1h 0m 0s,-90d), 1.5d >') );

SELECT srcontainsl(  spoint( '(1h 0m 0s,+89d)' ), scircle( '< (1h 0m 0s,+90d), 1.0d >') );

SELECT srcontainsl(  spoint( '(1h 0m 0s,-89d)' ), scircle( '< (1h 0m 0s,-90d), 1.0d >') );

-- should be "false"

SELECT srcontainsl(  spoint( '(1h 0m  0s,+1d    )' ), scircle( '< (2h 0m 0s,-1d),  1.5d >') );

SELECT srcontainsl(  spoint( '(3d 30m 0s,+1d    )' ), scircle( '< (0d 0m 0s, 0d),  3.5d >') );

SELECT srcontainsl(  spoint( '(1h 0m  0s,+88.99d)' ), scircle( '< (1h 0m 0s,+90d), 1.0d >') );

SELECT srcontainsl(  spoint( '(1h 0m  0s,-88.99d)' ), scircle( '< (1h 0m 0s,-90d), 1.0d >') );



-- spoislcontainsr( nt, scircle operato )r

-- should be "true"

SELECT slcontainsr( scircle( '< (1h 0m 0s,+1d),  1.5d >'), spoint( '(1h 0m 0s,+1d) ' ) );

SELECT slcontainsr( scircle( '< (1d 0m 0s, 0d),  3.5d >'), spoint( '(2d 0m 0s,+1d) ' ) );

SELECT slcontainsr( scircle( '< (1h 0m 0s,+90d), 1.5d >'), spoint( '(1h 0m 0s,+89d)' ) );

SELECT slcontainsr( scircle( '< (1h 0m 0s,-90d), 1.5d >'), spoint( '(1h 0m 0s,-89d)' ) );

SELECT slcontainsr( scircle( '< (1h 0m 0s,+90d), 1.0d >'), spoint( '(1h 0m 0s,+89d)' ) );

SELECT slcontainsr( scircle( '< (1h 0m 0s,-90d), 1.0d >'), spoint( '(1h 0m 0s,-89d)' ) );

-- should be "false"

SELECT slcontainsr( scircle( '< (2h 0m 0s,-1d),  1.5d >'), spoint( '(1h 0m  0s,+1d    )' ) );

SELECT slcontainsr( scircle( '< (0d 0m 0s, 0d),  3.5d >'), spoint( '(3d 30m 0s,+1d    )' ) );

SELECT slcontainsr( scircle( '< (1h 0m 0s,+90d), 1.0d >'), spoint( '(1h 0m  0s,+88.99d)' ) );

SELECT slcontainsr( scircle( '< (1h 0m 0s,-90d), 1.0d >'), spoint( '(1h 0m  0s,-88.99d)' ) );
