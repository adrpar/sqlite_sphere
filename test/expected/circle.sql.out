<(0.2727077 , 0.018180513) , 0.017453293>
<(0.2727077 , 0.018180513) , 0.026179939>
<(0.2727077 , 0.018180513) , 0.1>
<(0.2727077 , 0.018180513) , 1.5707963>
<(15.625d , 1.0416667d) , 1d>
<(15.625d , 1.0416667d) , 1.5d>
<(15.625d , 1.0416667d) , 5.729578d>
<(15.625d , 1.0416667d) , 90d>
<( 15d 37m 30s , + 1d  2m 30s) ,  1d  0m 0s>
<( 15d 37m 30s , + 1d  2m 30s) ,  1d 30m 0s>
<( 15d 37m 30s , + 1d  2m 30s) ,  5d 43m 46.480625s>
<( 15d 37m 30s , + 1d  2m 30s) , 90d  0m 0s>
<(  1h  2m 30s , + 1d  2m 30s) ,  1d  0m 0s>
<(  1h  2m 30s , + 1d  2m 30s) ,  1d 30m 0s>
<(  1h  2m 30s , + 1d  2m 30s) ,  5d 43m 46.480625s>
<(  1h  2m 30s , + 1d  2m 30s) , 90d  0m 0s>
1
1
19.8
13.0
20.0
0.499999999999953
0.0
87.0
178.0
( 15d 37m 30s , + 1d  2m 30s)
( 15d 37m 30s , +90d  0m 0s)
<(  0d  0m 0s , + 0d  0m 0s) ,  0d  0m 0s>
<(  0d  0m 0s , +90d  0m 0s) ,  0d  0m 0s>
<(  0d  0m 0s , -90d  0m 0s) ,  0d  0m 0s>
1
1
1
1
1
1
0
0
0
0
0
0
0
0
1
1
1
1
1
1
1
1
0
0
( 15d 30m 45s , + 1d  2m 3s)
( 15d 30m 45s , +90d  0m 0s)
19.8
13.0
20.0
0.499999999999953
0.0
87.0
178.0
1
1
1
1
0
0
0
0
1
1
1
1
0
0
0
0
1
1
1
1
1
1
0
0
0
0
1
1
1
1
1
1
0
0
0
0
